src/AssetsPage/AssetsTable/AssetsTable.test.jsx

```diff
@@ -24,31 +28,335 @@ const defaultProps = {
+          <AssetsTable
+            {...defaultProps}
+            updateSort={updateSortSpy}
+            assetsParameters={sortableAssetParameters}
+          />,
+        );
+
+        const sortButtons = wrapper.find('button').filterWhere(button => (
+          button.find('span > span .fa-sort').exists() ||
+          button.find('span > span .fa-sort-desc').exists() ||
+          button.find('span > span .fa-sort-asc').exists()
+        ));
+
+        expect(sortButtons).toHaveLength(3);
+
+        expect(sortButtons.at(testValues.buttonIndexToClick).filterWhere(button => (
+          button.text().includes(testValues.buttonToClickText)
+        ))).toHaveLength(1);
+
+        sortButtons.at(testValues.buttonIndexToClick).simulate('click');
+
+        expect(updateSortSpy).toHaveBeenCalledTimes(1);
+        expect(updateSortSpy).toHaveBeenLastCalledWith(
+          testValues.onSortColumnParameter,
+          testValues.onSortDirectionParameter,
+        );
+      });
+    });
+  });
+  describe('modal', () => {
+    let modal;
+
+    beforeEach(() => {
+      wrapper = mount(
+        <AssetsTable
+          {...defaultProps}
+        />,
+      );
+      modal = wrapper.find('[role="dialog"]');
+    });
+    it('renders', () => {
+      expect(modal).toHaveLength(1);
+    });
+    it('is closed by default', () => {
+      expect(modal.hasClass('modal-open')).toEqual(false);
+      expect(wrapper.state('modalOpen')).toEqual(false);
+    });
+    it('opens when trash button clicked; modalOpen state is true', () => {
+      const trashButtons = wrapper.find('button').filterWhere(button => button.hasClass('fa-trash'));
+      expect(trashButtons).toHaveLength(defaultProps.assetsList.length);
+
+      trashButtons.at(0).simulate('click');
+
+      expect(modal.hasClass('modal-open')).toEqual(true);
+      expect(wrapper.state('modalOpen')).toEqual(true);
+    });
+    it('closes when Cancel button clicked; modalOpen state is false', () => {
+      const trashButtons = wrapper.find('button').filterWhere(button => button.hasClass('fa-trash'));
+      trashButtons.at(0).simulate('click');
+
+      const closeButton = modal.find('button').filterWhere(
+        button => button.matchesElement(<button>Cancel</button>));
+      expect(closeButton).toHaveLength(1);
+
+      closeButton.at(0).simulate('click');
+
+      expect(wrapper.state('modalOpen')).toEqual(false);
+    });
+  });
+  describe('deleteAsset', () => {
+    it('calls deleteAsset function prop correctly', () => {
+      const deleteAssetSpy = jest.fn();
+
+      wrapper = mount(
+        <AssetsTable
+          {...defaultProps}
+          deleteAsset={deleteAssetSpy}
+        />,
+      );
+
+      const deleteButtons = wrapper.find('button');
+      const trashButtons = deleteButtons.filterWhere(button => button.hasClass('fa-trash'));
+      const deleteButton = deleteButtons.filterWhere(button =>
+        button.matchesElement(<button>Yes, delete.</button>));
```

Comment:
Curiosity - Can this be done with finding the buttons within the modal explicitly and then finding the one with `btn-primary` style?

Reply:
Actually below you do search from the modal buttons specifically: https://github.com/edx/studio-frontend/pull/41/files#diff-ca1fa9e3aa238807437edc787d4e8f9dR281 - still using the specific HTML tags.

Reply:
What's the advantage of that?

Reply:
This is more of a nit for the styling search -- the finding of buttons on the modal specifically than out of all the buttons in the off chance another "Yes, delete." button gets added anywhere else in the DOM. This is fine for now, but contextually it made sense to me to search for the appropriate button from within the modal here since that's where it lives (and you had done it that way in the other test, so I guess some uniformity too haha).

Reply:
Meaning something like this?

`wrapper.find('[role="dialog"]').filterWhere(button => button.matchesElement(<button>Yes, delete.</button>));`

Or do you prefer?

`wrapper.find('[role="dialog"]').filterWhere(button => button.hasClass('btn-primary'));`?

I feel like I prefer the first one since it selects for a delete specific button in the modal, as opposed to a primary button in the modal, which may not be the delete? I could go either way, though.